src/test/java/org/cactoos/func/ProcOfTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.func;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link ProcOf}.
+ *
+ * @author Mikhail Yakushin (driver733@me.com)
+ * @version $Id$
+ * @since 0.3
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 line)
+ */
+public final class ProcOfTest {
+
+    @Test
+    public void worksWithRunnable() throws Exception {
+        final String str = "test string";
+        final List<String> list = new ArrayList<>(1);
+        new ProcOf<String>(
+            new Runnable() {
+                @Override
+                public void run() {
+                    list.add(str);
+                }
+            }
+        ).exec("You can use any input in a Runnable.");
+        MatcherAssert.assertThat(
+            "Wrong result with a Runnable.",
+            list,
+            Matchers.contains(str)
+        );
```

Comment:
@driver733 How many times will ``list.add(...)`` be called? I guess once, can we have an assertion to check that the list has only one element?

Reply:
@amihaiemil This condition is already checked by the `Matcher.contains`
See [this](https://stackoverflow.com/a/38188509/2441104) for more info.